### ISSUE TITLE
[Bug]Missing factory connections on worldgen

### ISSUE THREAD

**Issue by 209CATrus** [jamespetts/simutrans-extended#257]
Well, name speaks for itself, i often get these cuts with big industries like oil refineries or car factories.
Possible reason - low industry number in worldgen settings. Perhaps industry builder takes industrial sector to agricultural sector ratio as priority over individual supply chains intergrity?

**Comment by 209CATrus** [comment]
Nope, my map doesn't have agricultural sector at all. So, another possible reasons:

- No suitable regions for farms on the map

- Large amount of heavy industries meets the demand for job places

**Comment by jamespetts** [comment]
I will need a reliable reproduction case in order to be able to look into a bug report: please see the bug report guidelines here: https://forum.simutrans.com/index.php/topic,13026.0.html

**Comment by 209CATrus** [comment]
I can't see any chenges to factories since i updated in 17th august, neither do i changed configs.

So, my worldgen parameters:
Seed 1085838446
1024*384

23
1
5
300
50

292
10
6

1950
No beginner mode

Climate:
-1
150
3

No hilly landscape
Cities ignore height

61
(20)
12
-1
-1
5
10
15
20

No no trees
No lakes

128
2
4096

It is not guaranteed to reproduce, but you should get two or so cases of missing connection in five attempts

**Comment by 209CATrus** [comment]
I didn't check it out myself yet, but it seems that recent commits made enough progress to close this issue. Thanks

**Comment by 209CATrus** [comment]
Okay, turns out it fails to replace incomplete supply chains. It just replaces same factory. 

Lack of available workforce? I've set number of industries above recommended value. 
Maybe because my world was flat, it had no biomes for complete iron/coal supply chains.

These two screenshots should provide you with both replication case and the issue:

Power station got new supply chain - consisting of Coal Fired Power Station. Exclusively.
Also, as you can see - steelworks got no chain update at all, having missing connection:

![simuexfactory1](https://user-images.githubusercontent.com/26685838/104948175-52aee880-59d6-11eb-9188-332d243acb26.png)

World parameters i've used:
![simuexfactory2](https://user-images.githubusercontent.com/26685838/104948179-55114280-59d6-11eb-949b-a8f51b945bca.png)

.
